src/method/query/response.rs

```diff
@@ -0,0 +1,237 @@
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::ops::Index;
+use std::slice::SliceIndex;
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use crate::Result;
+use crate::Value;
+
+/// A wrapper type around the list of results for the queries that were returned
+/// by the database.
+///
+/// Provides utility functions to access the result of one specific query, or if
+/// needed, all queries at once.
+///
+#[derive(Debug, Clone)]
+pub struct QueryResponse(Vec<QueryResult>);
+
+impl QueryResponse {
+    /// Constructs an empty [`QueryResponse`]
+    pub fn new() -> Self {
+        Self(Default::default())
+    }
+
+    /// Unwrap into the inner list of query results
+    pub fn into_inner(self) -> Vec<QueryResult> {
+        self.0
+    }
+
+    /// Returns a reference the result for the `n`-th query from the response. If
+    /// no result is found at this index then [None] is returned.
+    pub fn get_query(&self, n: usize) -> Option<&QueryResult> {
+        self.0.get(n)
+    }
+
+    /// Returns a reference to the result for the `n`-th query from the response.
+    ///
+    /// # Panics
+    /// Unlike the [`Self::get()`] function no bounding check is performed and may panic if
+    /// `n` is out of bounds.
+    pub fn get_query_unchecked(&self, n: usize) -> &QueryResult {
```

Comment:
This is covered by `impl Index<usize> for QueryResponse`, right? I feel like it isn't necessary. If one doesn't want to do `response[n]` they can still do `response.index(n)`.